Dev

AlonBru opened the pull request.

AlonBru committed:
overdue commit

AlonBru committed:
added  show hidden

AlonBru committed:
fixed search and hide

AlonBru committed:
Create TODO.md

AlonBru committed:
Update TODO.md

AlonBru committed:
lint fix and mark done

AlonBru committed:
added mailto

- added mailto link
-

AlonBru committed:
added sidebar ( to implement)

AlonBru committed:
Update .eslintrc.js

added rules to eslint

AlonBru committed:
Update App.css

made sidebar sidebarred

AlonBru committed:
Update App.js

added options state, display

AlonBru committed:
Update ShowButton.js

variable naming

AlonBru committed:
Update Ticket.js

variable naming

AlonBru committed:
options sidebar

added, WIP

AlonBru committed:
Update data.json

AlonBru committed:
Update Ticket.js

updated for new ticket rendering method

AlonBru committed:
implementing filtering options

added filter closed tickets
added time range selector

AlonBru committed:
added filtering by time created

AlonBru committed:
filter by labels

fixed some naming issues,
added filter by labels
generalized Search

AlonBru committed:
Update App.js

moved hideTicket up cause i dont dig hoisting

AlonBru committed:
menuclose

added close/open to options menu
\

AlonBru committed:
streamlined code

removed duplicity

AlonBru committed:
CSSing and lintings

AlonBru committed:
added color

to implement choosing colors

AlonBru committed:
Update TODO.md

AlonBru committed:
Beautification

AlonBru committed:
fixed label buttons

 in tickets

GuySerfaty committed:
Update README.md

GuySerfaty committed:
Update final.test.js

AlonBru committed:
fixes& README

AlonBru committed:
Merge branch 'DEV' of https://github.com/AlonBru/Ticket-Manager into DEV

AlonBru committed:
Update README.md

AlonBru committed:
z

AlonBru committed:
my tests

AlonBru committed:
tests

AlonBru committed:
Update mytests.test.js

AlonBru committed:
Revert "Update mytests.test.js"

This reverts commit a63507ea70f018b428689cbe576316764a061fd2.

AlonBru committed:
removed tests

my tests didnt work in github

Cyber4sPopo commented on client/src/App.js:
```diff
@@ -1,12 +1,210 @@
+  const toggleMenu = () => {
+    const changedOptions = { ...options };
+    changedOptions.displayMenu = !changedOptions.displayMenu;
+    setOptions(changedOptions);
```
When set state depends on old state better use callback

AlonBru replied:
what do you mean? use a callback for what?

Cyber4sPopo commented on client/src/App.js:
```diff
@@ -1,12 +1,210 @@
+    setTicketsToDisplay(copyOfTickets);
+  }
+  const labelClick = (e) => {
+    if (e.target.className === 'label') {
```
Better to build it in a way only label use `labelClick` method

AlonBru replied:
event delegation is not considered good practice in react?

Cyber4sPopo commented on client/src/App.js:
```diff
@@ -1,12 +1,210 @@
+
+      <div id="ticketCounterContainer">
+        <span id="ticketCounter">
+          {renderTickets().length}
```
really bad, you render all just for getting the length?  try to call renderTickets once + I know this how react act in their tutorial but I can't see why to move out this logic to an external

AlonBru replied:
@Cyber4sPopo 
it is done so i always have the updated count displayed in the counter.
I tried making a variable ticketCount which the renderTickets function sets when it is called, but since the counter is rendered before the tickets are ( before the function is called)  it means it isn't set when I need it. it can be fixed with moving the counter down in the code and then back up with some css magic...
do you think it would be better to make another function which does basically the same as renderTickets but doesn't render in the end in order to get the ticket count?

AlonBru replied:
i changed it so something else.

Cyber4sPopo commented on client/src/App.js:
```diff
@@ -1,12 +1,210 @@
+      />
+    ));
+  }
+  function unHideTickets() {
```
reset / restore isn't better ?

AlonBru replied:
I find this more specific

Cyber4sPopo commented on client/src/App.js:
```diff
@@ -1,12 +1,210 @@
+        </span>
+        <ShowButton
+          hiddenTickets={ticketsHidden.length}
+          function={unHideTickets}
```
bad naming **function** ?!

Cyber4sPopo commented on client/src/components/Sidebar.js:
```diff
@@ -0,0 +1,119 @@
+        <br />
+        <input type="button" value="Cheerful Pink" style={{ background: '#ffb6c1', color: 'white' }} onClick={changeColour} />
+        <br />
+        <input type="button" value="Chilled Sangria" style={{ background: '#8b0000', color: 'white' }} onClick={changeColour} />
```
a lot of duplicate code ! maybe create new component called `<ColorOption value="Cheerful Pink" style={{ background: '#ffb6c1', color: 'white' }} />`

Cyber4sPopo commented on client/src/components/Ticket.js:
```diff
@@ -0,0 +1,78 @@
+import Labels from './Label';
+
+const Ticket = (props) => {
+    const {data,labelClick} = props
```
eslint !!!

Cyber4sPopo commented on server/app.js:
```diff
@@ -1,5 +1,55 @@
+    const tickets = JSON.parse(await fs.readFile("./data.json",'utf-8'));
+    const changedArray= tickets.map(ticket => {
+        if(ticket.id===req.params.ticketId){
+            ticket.done=true;        }
```
eslint

Cyber4sPopo commented on server/app.js:
```diff
@@ -1,5 +1,55 @@
+    await fs.writeFile('./data.json',json)
+    res.send({updated:true})
+})
+app.post('/api/tickets/:ticketId/undone', async (req, res)=> {
```
2 entry point the same?

AlonBru committed:
Update app.js

- removed duplicate entry point
- eslinted

trodelgal commented on client/src/components/Label.js:
```diff
@@ -0,0 +1,13 @@
+
+const Labels = (props) => {
+  const { data } = props;
+  if (data) {
```
you can use defaultProps instead of condition.

AlonBru committed:
Additions and stuff

- #4 fixed you can mark tickets as done while searching
- #3 fixed
- #2 fixed
- greyed out buttons in done ticket
- disabled mail in done ticket
- fixed ticket count relying on rerendering all tickets function
- relegated color options into ne component
-added restore label button to menu and status bar

Cyber4sPopo commented:
make the page fit without horizontal scroll bar 
![image](https://user-images.githubusercontent.com/69596366/91935134-a950aa00-ecf5-11ea-8226-8105dbf275bc.png)

Cyber4sPopo commented on client/src/App.js:
```diff
@@ -1,12 +1,210 @@
+      filteredTickets = filteredTickets.filter(filterByDone);
+    }
+    if (timeRange.active) { // filter tickets by creation time
+      const filterByTime = (ticket) => {
```
get this function outside of a function, this function might be reusable at the future.

AlonBru committed:
improvements

- made favcolor work better
-

AlonBru committed:
for presentation
